triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala

```diff
@@ -57,30 +60,33 @@ abstract class TriggerMultiProcessSimulation extends AsyncWordSpec with Abstract
       Behaviors
         .supervise[Message] {
           Behaviors.setup { context =>
+            var processSimulation: Option[ActorRef[Unit]] = None
+
             context.log.info(s"Simulation will run for ${simulationConfig.simulationDuration}")
             context.self ! StartSimulation
 
             Behaviors.logMessages {
-              Behaviors.receiveMessage {
-                case StartSimulation =>
-                  triggerMultiProcessSimulation.transformMessages {
-                    case StartSimulation =>
-                      ()
-                    case StopSimulation =>
-                      throw TriggerSimulationFailure(
-                        new TimeoutException(
-                          s"Simulation stopped after ${simulationConfig.simulationDuration}"
-                        )
-                      )
-                  }
-
-                case StopSimulation =>
-                  throw TriggerSimulationFailure(
-                    new TimeoutException(
+              Behaviors
+                .receiveMessage[Message] {
+                  case StartSimulation =>
+                    processSimulation =
+                      Some(context.spawn(triggerMultiProcessSimulation, "simulation-controller"))
+                    processSimulation.foreach(context.watch)
+                    Behaviors.same
+
+                  case StopSimulation =>
+                    context.log.info(
                       s"Simulation stopped after ${simulationConfig.simulationDuration}"
                     )
-                  )
-              }
+                    processSimulation.foreach(context.stop)
+                    processSimulation = None
+                    simulationTerminatedNormally.success(())
+                    Behaviors.stopped
+                }
+                .receiveSignal { case (_, ChildFailed(_, cause)) =>
```

Comment:
If any supervised (i.e. child) actor fails, we'll get this signal and the reason it failed.